api/ask_astro/chains/custom_llm_filter_prompt.py

```diff
@@ -0,0 +1,39 @@
+from langchain.retrievers.document_compressors.chain_filter_prompt import (
+    prompt_template,
+)
+from langchain_core.output_parsers import BaseOutputParser
+from langchain_core.prompts import PromptTemplate
+
+
+class CustomBooleanOutputParser(BaseOutputParser[bool]):
```

Comment:
Note: this is a changed implementation from langchain. The original code looked like this https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/output_parsers/boolean.py